src/include/simeng/arch/aarch64/helpers/sve.hh

```diff
@@ -246,13 +249,15 @@ class sveHelp {
    * Returns single value of type uint64_t. */
   // TODO : Add support for patterns
   template <typename T>
-  static uint64_t sveDec_scalar(
+  static int64_t sveDec_scalar(
       std::array<RegisterValue, Instruction::MAX_SOURCE_REGISTERS>& operands,
       const simeng::arch::aarch64::InstructionMetadata& metadata,
       const uint16_t VL_bits) {
-    const uint64_t n = operands[0].get<uint64_t>();
+    const int64_t n = operands[0].get<int64_t>();
     const uint8_t imm = static_cast<uint8_t>(metadata.operands[1].imm);
-    return (n - ((VL_bits / (sizeof(T) * 8)) * imm));
+    const uint16_t elems =
+        AuxFunc::sveGetPattern(metadata.operandStr, sizeof(T) * 8, VL_bits);
+    return (n - (elems * imm));
```

Comment:
is it right that a signed `n` is calculated with unisigned `elems * imm` here ? not sure what this function is doing so am naive